shared/array.c

```diff
@@ -66,10 +66,14 @@ int array_append(struct array *array, const void *element)
 int array_append_unique(struct array *array, const void *element)
 {
 	void **itr = array->array;
-	void **itr_end = itr + array->count;
-	for (; itr < itr_end; itr++)
-		if (*itr == element)
-			return -EEXIST;
+
+	if (itr) {
```

Comment:
that's a weird condition for what this is fixing. We should do nothing if array->count is 0, rather than do nothing if itr is NULL, which it never is.